scipy/optimize/tests/test__differential_evolution.py

```diff
@@ -259,23 +259,45 @@ def test_best_solution_retrieval(self):
         result = solver.solve()
         assert_equal(result.x, solver.x)
 
+    def test_intermediate_result(self):
+        bounds = [(0, 2), (0, 2)]
+        visited = [False]
+        def callback(intermediate_result):
+            visited[0] = True
+            assert intermediate_result.population.ndim == 2
+            assert intermediate_result.population.shape[1] == 2
+
+        result = differential_evolution(rosen, bounds, callback=callback)
+        assert result.success
+        assert visited[0]
+
     def test_callback_terminates(self):
         # test that if the callback returns true, then the minimization halts
         bounds = [(0, 2), (0, 2)]
-        expected_msg = 'callback function requested stop early by returning True'
-
+        expected_msg = 'callback function requested stop early'
         def callback_python_true(param, convergence=0.):
             return True
 
-        result = differential_evolution(rosen, bounds, callback=callback_python_true)
+        result = differential_evolution(
+            rosen, bounds, callback=callback_python_true
+        )
         assert_string_equal(result.message, expected_msg)
 
+        # if callback raises StopIteration then solve should be interrupted
+        # with no polishing
```

Comment:
It says in the documentation that polishing is still carried out, right?

Reply:
Polishing still occurs, I've removed 287